### PR TITLE
feature: add a parse parameter degree_as_nlabel for pytorch-gin demo

### PR DESCRIPTION
**Pull request by miziha-zp**
## Description

1. the original `readme.md` for gin seems to be a little hard to reproduce the result in the paper for people that didn't notice the details about the implementation, so add some hints and demo scripts. #3570 

2. `GINDataset` has an initial parameter `degree_as_nlabel`, but `main.py` did not expose an argument for it, so I add it.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR
- [x] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/pytorch/gin/README.md [hunk 1]
```diff
@@ -28,15 +28,22 @@ An experiment on the GIN in customized settings can be run with
 python main.py [--device 0 | --disable-cuda] --dataset COLLAB \
                --graph_pooling_type max --neighbor_pooling_type sum
 ```
+hint: **following the settings like the paper** is essential to reproduce the result reported.
+
+add `--degree_as_nlabel` to  use one-hot encodings of node degrees as node feature vectors
```
Comment (mufeili) [inline examples/pytorch/gin/README.md:33]: extra empty space between `to` and `use`

File: examples/pytorch/gin/README.md [hunk 2]
```diff
@@ -28,15 +28,22 @@ An experiment on the GIN in customized settings can be run with
 python main.py [--device 0 | --disable-cuda] --dataset COLLAB \
                --graph_pooling_type max --neighbor_pooling_type sum
 ```
+hint: **following the settings like the paper** is essential to reproduce the result reported.
```
Comment (mufeili) [inline examples/pytorch/gin/README.md:31]: This line is not needed. You just need to clearly list the commands for different datasets in the "Results" section.

File: examples/pytorch/gin/README.md [hunk 3]
```diff
@@ -28,15 +28,22 @@ An experiment on the GIN in customized settings can be run with
...
+
+add `--degree_as_nlabel` to  use one-hot encodings of node degrees as node feature vectors
 
 Results
 -------
 
 Run with following with the double SUM pooling way:
```
Comment (mufeili) [inline examples/pytorch/gin/README.md:38]: "Run with following with the double SUM pooling way:" This does not hold for all datasets. I think it's better to simply list the commands for all datasets, one command per dataset.

**Review by mufeili — CHANGES_REQUESTED** [review]
Some minor comments

File: examples/pytorch/gin/README.md (hunk 3 above)
Comment (miziha-zp) [reply examples/pytorch/gin/README.md:38]: I will update this section after testing every dataset.

File: examples/pytorch/gin/README.md [hunk 4]
```diff
@@ -28,18 +28,58 @@ An experiment on the GIN in customized settings can be run with
...
                --graph_pooling_type max --neighbor_pooling_type sum
 ```
+add `--degree_as_nlabel` to use one-hot encodings of node degrees as node feature vectors
 
 Results
 -------
 
 Run with following with the double SUM pooling way:
-(tested dataset: "MUTAG"(default), "COLLAB", "IMDBBINARY", "IMDBMULTI")
+(tested dataset: "MUTAG"(default), "COLLAB", "IMDBBINARY", "IMDBMULTI"), results may **fluctuate**, due to random factors and the relatively small data set.
```
Comment (mufeili) [inline examples/pytorch/gin/README.md:37]: You can simply remove "Run with following with the double SUM pooling way: (tested dataset: "MUTAG"(default), "COLLAB", "IMDBBINARY", "IMDBMULTI"),"

File: examples/pytorch/gin/README.md [hunk 5]
```diff
@@ -35,16 +35,51 @@ Results
...
+                --graph_pooling_type sum --neighbor_pooling_type sum --filename PTC.txt
+
+python main.py --dataset NCI1 --device 0  \
+                --graph_pooling_type sum --neighbor_pooling_type sum --filename NCI1.txt
 
+python main.py --dataset PROTEINS --device 0  \
+                --graph_pooling_type sum --neighbor_pooling_type sum --filename PROTEINS.txt
+
+# 5 social network datasets setting graph_pooling_type=mean, for the REDDIT datasets, we set all node feature vectors to be the same 
+# (thus, features here are uninformative); for the other social graphs, we use one-hot encodings of node degrees.
```
Comment (mufeili) [inline examples/pytorch/gin/README.md:55]: A minor issue: people use the term "social networks" more often

**Pull request merged**